src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
+                relatives_with_guardian.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                relatives_with_guardian_ids.index, "guardian_1"
+            ] = relatives_with_guardian_ids
+
+        # Children are relative of reference person with no age bound relative(s) - yellow box
+        relative_ids = (
+            household_structure.loc[
+                child_relative_of_rp_idx.drop(
+                    relatives_with_guardian.index.unique("child_id"), level="child_id"
+                ).intersection(rp_idx)
+            ]
+            .reset_index()
+            .set_index("child_id")["person_id"]
+        )
+        pop.loc[relative_ids.index, "guardian_1"] = relative_ids
+        # Assign guardian to spouse/partner of reference person
+        relative_with_rp_partner_guardian_ids = household_structure.loc[
+            child_relative_of_rp_idx.drop(
+                relatives_with_guardian.index.unique("child_id"), level="child_id"
+            ).intersection(partners_idx)
+        ]
+        if len(relative_with_rp_partner_guardian_ids) > 0:
+            relative_with_rp_partner_guardian_ids = (
+                relative_with_rp_partner_guardian_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                relative_with_rp_partner_guardian_ids.index, "guardian_2"
+            ] = relative_with_rp_partner_guardian_ids
+
+        # Children are reference person and have a parent in household - green box
+        child_rp_with_parent = household_structure.loc[
+            child_rp_idx.intersection(parents_of_rp_idx)
+        ]
+        if len(child_rp_with_parent) > 0:
+            child_rp_with_parent_ids = (
+                child_rp_with_parent.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[child_rp_with_parent_ids.index, "guardian_1"] = child_rp_with_parent_ids
+
+        # Child is reference person with no parent, assign to age bound relative - blue box
+        child_rp_with_relative_ids = household_structure.loc[
+            child_rp_idx.drop(child_rp_with_parent.index.unique("child_id"), level="child_id")
+            .intersection(relatives_of_rp_idx)
+            .intersection(age_bound_idx)
+        ]
+        if len(child_rp_with_relative_ids) > 0:
+            child_rp_with_relative_ids = (
+                child_rp_with_relative_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                child_rp_with_relative_ids.index, "guardian_1"
+            ] = child_rp_with_relative_ids
+
+        # Child is not related to and is not reference person, assign to age bound non-relative - blurple box
```

Comment:
is this blue-purple? that box appears to say assign any non-relative over 18. should it be over 18, or age-bound? if i was understanding correctly the "age-bound" was between 20-45 years older, right?

Reply:
oh wait i was misreading; thats the next one